### DISCUSSION TITLE
Special symbol ~ conversion

### DISCUSSION THREAD

**Discussion by GardenHamster** [aaubry/YamlDotNet#936] [Q&A] [answered]
symbol ~

```csharp
List<string> prefixs = new List<string> { "~","#" };
var serializer = new SerializerBuilder();
var yamlContent = serializer.Build().Serialize(prefixs);
Console.WriteLine(yamlContent);
```

output
``` yml
- ~
- '#'
```

the symbol ~ cannot be deserialized

is there any way to achieve the following effect
``` yml
- '~'
- '#'
```
or
``` yml
['~','#']
```

**Comment by EdwardCooke** [comment]
There is. On your serializerbuilder call WithQuoteNecessaryStrings before Build. That should do what you want.

**Reply by GardenHamster** [reply]
Thanks~

**Comment by EdwardCooke** [comment, marked as answer]
Try using withquotenecessarystrings on your serializerbuilder. That should automatically add quotes around scalars that are numbers only. Along with other special values.
